File: backend/src/main/java/com/leadfinder/dto/SearchRequest.java
package com.leadfinder.dto;

import jakarta.validation.constraints.NotBlank;

public class SearchRequest {

    @NotBlank(message = "Name is required")
    private String name;

    private String company;
    private String title;

    public SearchRequest() {
    }

    public SearchRequest(String name, String company, String title) {
        this.name = name;
        this.company = company;
        this.title = title;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: linkedin/lib/models/search_request.dart
class SearchRequest {
  final String name;
  final String? company;
  final String? title;

  SearchRequest({required this.name, this.company, this.title});

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'company': company,
      'title': title,
    };
  }
}
